src/transformState.js

```diff
@@ -5,7 +5,39 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    const actionType = action.type;
+
+    switch (actionType) {
+      case 'addProperties':
+        const extraData = action.extraData;
+
+        Object.assign(state, extraData);
+
+        break;
+
+      case 'removeProperties':
+        const keysToRemove = action.keysToRemove;
+
+        keysToRemove.forEach(key => {
+          delete state[key];
+        });
+
+        break;
+
+      case 'clear':
+        const stateKeys = Object.keys(state);
+
+        stateKeys.forEach(key => {
```

Comment:
No need to declare variables to be used only once on the next line 
```suggestion
        Object.keys(state).forEach(key => {
```